tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -1422,6 +1448,85 @@ int trace_ret_##name(void *ctx)                                         \
     return trace_ret_generic(ctx, id, types, tags, &args, ret);         \
 }
 
+static __always_inline void network_to_host_v4(net_conn_v4_t *net_details)
+{
+    net_details->local_address = __bpf_ntohl(net_details->local_address);
+    net_details->local_port = __bpf_ntohs(net_details->local_port);
+    net_details->remote_address = __bpf_ntohl(net_details->remote_address);
+    net_details->remote_port = __bpf_ntohs(net_details->remote_port);
+}
+
+static __always_inline int get_network_details_from_sock_v4(struct sock *sk, net_conn_v4_t *net_details, int peer)
+{
+    struct inet_sock *inet = inet_sk(sk);
+
+    u32 addr = 0;
+    bpf_probe_read(&addr, sizeof(addr), &inet->inet_rcv_saddr);
+    if ( !addr ) {
+        bpf_probe_read(&addr, sizeof(addr), &inet->inet_saddr);
+    }
+
+    if ( peer ) {
+
+        bpf_probe_read(&net_details->local_address, sizeof(net_details->local_address), &inet->inet_daddr);
```

Comment:
kernel structs (inet_sock in this case) may change between different kernel versions.
For this reason, we dereference structs using helper functions which are located in the "KERNEL VERSION DEPENDANT HELPER FUNCTIONS" section of the bpf source code.
Any field we read from a kernel struct (e.g. inet->inet_daddr) can have its own helper.

Other than that, you can use the `READ_KERN` helper instead of `bpf_probe_read`